src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -452,13 +467,28 @@ private async Task<TestResult> ExecuteInternalAsync(object?[]? arguments, Cancel
                                 try
                                 {
                                     object? invokeResult = TestMethod.GetInvokeResult(_classInstance, arguments);
-                                    if (invokeResult is Task task)
+                                    if (invokeResult is null)
+                                    {
+                                        // Do nothing.
+                                        // Don't remove this if condition as we don't want to go in the "else" in this case.
+                                    }
+                                    else if (invokeResult is Task task)
                                     {
                                         await task;
                                     }
-                                    else if (invokeResult is ValueTask valueTask)
+                                    else
                                     {
-                                        await valueTask;
+                                        await TryAwaitValueTaskAsync(invokeResult);
+
+                                        // Avoid loading System.Threading.Tasks.Extensions if not needed.
+                                        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
I assume here we rely on the behavior of JITter, that will load types mentioned in a method just before entering the method. Is that correct? Might be worth explaining that a little futher in the comment here (and the one below)?

Reply:
Thanks for the info. This is good to know!